tests/conversion_test.py

```diff
@@ -546,6 +547,32 @@ def test_pytket_qir_20(profile: QIRProfile) -> None:
     run_qir_gen_and_check(circ, "test_pytket_qir_20", profile=profile)
 
 
+@pytest.mark.parametrize(
+    "profile",
+    [
+        QIRProfile.ADAPTIVE,
+        QIRProfile.AZUREADAPTIVE,
+        QIRProfile.PYTKET,
+        QIRProfile.ADAPTIVE_CREGSIZE,
+    ],
+)
+def test_pytket_qir_21(profile: QIRProfile) -> None:
+    # test classical expression handling with register predicates
+    circ = Circuit(2)
+    a = circ.add_c_register("a", 3)
+    b = circ.add_c_register("b", 3)
+    c = circ.add_c_register("c", 1)
+
+    circ.add_clexpr(*wired_clexpr_from_logic_exp(reg_eq(a, b), c))  # type: ignore
```

Comment:
Why do we have to type ignore this? Is this not typed in pytket?

Reply:
For the same reason that we have to [here](https://github.com/CQCL/pytket-qir/blob/6df5e1b633e60e23d660b29defe6fe6520681181/tests/conversion_test.py#L193) (this test was based on that one). I think it may be to do with implicitly casting a `BitRegister` to a `list`.